src/modules/commander/Commander.cpp

```diff
@@ -2081,6 +2081,9 @@ Commander::run()
 					   ) {
 						print_reject_arm("Not arming: Switch to a manual mode first");
 
+					} else if (vtol_status.vtol_in_trans_mode) {
+						print_reject_arm("Not arming: Vehicle is in transition state");
```

Comment:
This chunk of code only applies to arming via sticks. Let's push the logic into the pre-arm checks so it's caught in any arming attempt. 

https://github.com/PX4/Firmware/blob/master/src/modules/commander/Arming/PreFlightCheck/checks/preArmCheck.cpp#L123

You can also use `vehicle_status. in_transition_mode` instead of vtol_vehicle_status.

https://github.com/PX4/Firmware/blob/bd98a4eebbee750a4db007b03bcd1e40896cb5d5/msg/vehicle_status.msg#L77

Reply:
Thanks for checking it.
I've moved it to the pre-arm check and tested it on the bench when arming through QGC and the MAVLink Console.